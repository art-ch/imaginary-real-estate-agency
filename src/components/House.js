import React, { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';

import HouseWrapper from '../styled/House';
import { Container, Row, Col, Image, Button, Spinner } from 'react-bootstrap';

import { useDispatch, useSelector } from 'react-redux';
import { useParams, useNavigate } from 'react-router-dom';
import { getSingleHouse, resetSingleHouse } from '../redux/ducks/getHouse';
import { priceHandler } from '../utils';

const House = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getSingleHouse(id));
    return () => {
      dispatch(resetSingleHouse(id));
    };
    // eslint-disable-next-line
  }, []);

  const currentHouse = useSelector((state) => state.getHouseReducer.house);
  const { house, gallery } = currentHouse;

  if (house && gallery) {
    const { image, title, price, address, description, name } = house;
    return (
      <HouseWrapper>
        <Container className="img-container">
          <Image src={image} alt="" />
        </Container>
        <Container className="my-2 g-4">
          <h1>{title}</h1>
          <p className="h4">{priceHandler(price)}</p>
          <p className="h6">{address}</p>
          <p>{description}</p>
          <p className="text-uppercase fst-italic">{name}</p>
          <Container className="container-sm p-0 mb-3">
            <Row className="flex-column g-2">
              <Col className="w-100 d-flex">
                <Button variant="outline-primary" className="me-2 w-100">
                  Contact Agent
                </Button>
                <Button className="w-100">Take a Tour</Button>
              </Col>
              <Col>
                <Button
                  className="w-100"
                  variant="warning"
                  onClick={() => navigate('/')}
                >
                  Back
                </Button>
              </Col>
            </Row>
          </Container>
          <Container className="p-0">
            <Row>
              <h2 className="text-center">Gallery</h2>
              {gallery.map((image, id) => {
                return (
                  <Col className="mb-3 col-12 col-sm-4 col-xl-3" key={id}>
                    <a href={image}>
                      <img src={image} alt="" />
                    </a>
                  </Col>
                );
              })}
            </Row>
          </Container>
        </Container>
      </HouseWrapper>
    );
  }
  return (
    <div className="mt-3 text-center">
      <Spinner className="center-block" animation="border" variant="primary" />
    </div>
  );
};

export default House;
