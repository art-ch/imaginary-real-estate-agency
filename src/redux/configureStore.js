import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import getRealEstateReducer from './ducks/getRealEstate';
import getHouseReducer from './ducks/getHouse';
import { watcherSaga } from './sagas/rootSaga';

const reducer = combineReducers({ getRealEstateReducer, getHouseReducer });

const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

const store = createStore(reducer, {}, applyMiddleware(...middleware));

sagaMiddleware.run(watcherSaga);

export default store;
