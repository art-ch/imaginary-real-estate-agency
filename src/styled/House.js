import styled from 'styled-components';

const HouseWrapper = styled.main`
  margin: 0 auto;
  .container {
    max-width: 1000px;
  }
`;

export default HouseWrapper;
