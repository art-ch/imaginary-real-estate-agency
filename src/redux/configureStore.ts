import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import getRealEstateReducer from './ducks/getRealEstate';
import getHouseReducer from './ducks/getHouse';
import { watcherSaga } from './rootSaga';

const reducer = combineReducers({ getRealEstateReducer, getHouseReducer });

const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

const store = createStore(reducer, {}, applyMiddleware(...middleware));

sagaMiddleware.run(watcherSaga);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
