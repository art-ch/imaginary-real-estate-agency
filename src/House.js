import React, { useContext } from 'react';
import { Context } from './Context';
import styled from 'styled-components';
import { Container, Image, Button, Carousel } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

const House = () => {
  const gallery = [
    'https://images.pexels.com/photos/164338/pexels-photo-164338.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/1457842/pexels-photo-1457842.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/1080721/pexels-photo-1080721.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/90317/pexels-photo-90317.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/1454804/pexels-photo-1454804.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/5252546/pexels-photo-5252546.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
    'https://images.pexels.com/photos/13975/pexels-photo-13975.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
  ];
  const { priceHandler } = useContext(Context);
  return (
    <Wrapper>
      <Container className="img-container">
        <Image
          src="https://images.pexels.com/photos/164338/pexels-photo-164338.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
          alt=""
        />
      </Container>
      <Container className="my-2 g-4">
        <h1>Townhome</h1>
        <p className="h4">{priceHandler(289900)}</p>
        <p className="h6">1933 Frum Street, Nashville, TN 37212</p>
        <p>
          Broker ipsum dolor sit amet, consectetuer stainless steel elit. Trendy
          shops commodo ligula eget dolor. Cum sociis luxury penatibus et magnis
          dis modern interior montes, nascetur soaring views mus. Location
          location location quam felis, ultricies nec, pellentesque
          rent-controlled tenant in place, pretium quis, sem.
        </p>
        <Carousel className="my-5">
          {gallery.map((image) => {
            return (
              <Carousel.Item>
                <Container className="img-container">
                  <img className="w-100 h-100" src={image} alt="" />
                </Container>
              </Carousel.Item>
            );
          })}
        </Carousel>
        <p className="text-uppercase fst-italic">Tiffany Marin</p>
        <Button>Contact Agent</Button>
      </Container>
    </Wrapper>
  );
};

export default House;

const Wrapper = styled.main`
  margin: 0 auto;
  .img-container {
    height: 300px;
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
  .container {
    max-width: 1000px;
  }

  @media (min-width: 767px) {
    .img-container {
      height: 400px;
    }
  }

  @media (min-width: 1200px) {
    .img-container {
      height: 570px;
    }
  }
`;
