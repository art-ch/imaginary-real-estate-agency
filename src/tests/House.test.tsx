import { Provider } from 'react-redux';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import House from '../components/House';

const mockStore = configureStore([]);

describe('<House />', () => {
  let store;
  let component: any;

  beforeEach(() => {
    store = mockStore({
      getHouseReducer: {
        house: {
          house: {
            id: '1',
            title: 'House1',
            price: 228000,
            address: '1997 Yates Ave, Montebello, CA',
            image: 'image',
            description: 'desc',
            name: 'Ronnie Polla'
          },
          gallery: ['img1', 'img2', 'img3']
        }
      }
    });

    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Routes>
            <Route path="/" element={<House />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
