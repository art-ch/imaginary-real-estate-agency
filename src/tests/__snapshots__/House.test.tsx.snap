// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<House /> should match snapshot 1`] = `
<main
  className="sc-bdvvtL frHYMz"
>
  <div
    className="img-container container"
  >
    <img
      alt=""
      className=""
      src="image"
    />
  </div>
  <div
    className="my-2 g-4 container"
  >
    <h1>
      House1
    </h1>
    <p
      className="h4"
    >
      $228,000
    </p>
    <p
      className="h6"
    >
      1997 Yates Ave, Montebello, CA
    </p>
    <p>
      desc
    </p>
    <p
      className="text-uppercase fst-italic"
    >
      Ronnie Polla
    </p>
    <div
      className="container-sm p-0 mb-3 container"
    >
      <div
        className="flex-column g-2 row"
      >
        <div
          className="w-100 d-flex col"
        >
          <button
            className="me-2 w-100 btn btn-outline-primary"
            disabled={false}
            type="button"
          >
            Contact Agent
          </button>
          <button
            className="w-100 btn btn-primary"
            disabled={false}
            type="button"
          >
            Take a Tour
          </button>
        </div>
        <div
          className="col"
        >
          <button
            className="w-100 btn btn-warning"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
        </div>
      </div>
    </div>
    <div
      className="p-0 container"
    >
      <div
        className="row"
      >
        <h2
          className="text-center"
        >
          Gallery
        </h2>
        <div
          className="mb-3 col-12 col-sm-4 col-xl-3 col"
        >
          <a
            href="img1"
          >
            <img
              alt=""
              src="img1"
            />
          </a>
        </div>
        <div
          className="mb-3 col-12 col-sm-4 col-xl-3 col"
        >
          <a
            href="img2"
          >
            <img
              alt=""
              src="img2"
            />
          </a>
        </div>
        <div
          className="mb-3 col-12 col-sm-4 col-xl-3 col"
        >
          <a
            href="img3"
          >
            <img
              alt=""
              src="img3"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</main>
`;
